File: test/graphql/fetch.go
package graphql

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

const URL string = "http://localhost:8080/query"

func Fetch(data interface{}) []byte {
	client := &http.Client{}
	URL := "http://localhost:8080/query"
	payloadBuf := new(bytes.Buffer)
	json.NewEncoder(payloadBuf).Encode(data)
	req, err := http.NewRequest("POST", URL, payloadBuf)
	if err != nil {
		fmt.Print(err.Error())
	}
	req.Header.Add("Accept", "application/json")
	req.Header.Add("Content-Type", "application/json")
	resp, err := client.Do(req)
	if err != nil {
		fmt.Print(err.Error())
	}
	defer resp.Body.Close()
	bodyBytes, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Print(err.Error())
	}
	return bodyBytes
}

File: jest/fetch.js
const nodeFetch = require('node-fetch');


 export const fetch = async ({ query, variables }) => {
    expect("\n==================================================")
        .toMatchSnapshot();
    expect({ query, variables }).toMatchSnapshot();
    const resp = await nodeFetch('http://localhost:8080/query', {
            method: 'post',
            body: JSON.stringify({query, variables}),
            headers: { 'Content-Type': 'application/json' },
    })
    .then(resp => resp.json());
    return resp;
}